packages/wdio-junit-reporter/src/index.ts

```diff
@@ -322,7 +322,9 @@ class JunitReporter extends WDIOReporter {
 
         if (suiteKeys.length === 0) {
             const error = this.runnerStat?.error ?? 'No tests found'
-            return builder.testCase().failure(error)
+            if (!isCucumberFrameworkRunner || (isCucumberFrameworkRunner && type === 'feature')) {
```

Comment:
Can you elaborate why we need this if statement? Maybe to help have contributors understand what this means, a comment may make sense here.

Reply:
No, this makes total sense. Let's just add a comment help the next person understand this context because it is almost impossible to do so without this. Just add a comment with the appropriate context information.